web/@linera/client/src/lib.rs

```diff
@@ -408,9 +425,14 @@ impl Application {
     #[wasm_bindgen]
     // TODO(#14) allow passing bytes here rather than just strings
     // TODO(#15) a lot of this logic is shared with `linera_service::node_service`
-    pub async fn query(&self, query: &str, block_hash: Option<String>) -> JsResult<String> {
+    pub async fn query(&self, query: &str, options: JsValue) -> JsResult<String> {
         tracing::debug!("querying application: {query}");
-        let chain_client = self.client.default_chain_client().await?;
+        let QueryOptions { block_hash, owner } =
+            serde_wasm_bindgen::from_value::<Option<_>>(options)?.unwrap_or_default();
+        let mut chain_client = self.client.default_chain_client().await?;
+        if let Some(owner) = owner {
+            chain_client.set_preferred_owner(owner);
+        }
```

Comment:
With this, if client receives an epoch-changing event it'd still ask user to sign manually (with the preferred owner). Instead, the `options.owner` should be set as preferrable only for the duration of the `query` and reset it back to the previous one at the end of it. Otherwise it's nullifying the potential autosigner.

Reply:
`set_preferred_owner` calls `update_wallet_from_client` which updates the wallet.

Reply:
Ah nvm... There are two methods with the same name – `set_preferred_owner` – one on `ChainClient` and another on `ClientContext` 🤦 and the former (as is the case here) only changes for the instance as you say..